### PR TITLE
Always register the metadata endpoint

### PR DESCRIPTION
**Pull request by EngHabu**
Signed-off-by: Haytham Abuelfutuh <haytham@afutuh.com>

# TL;DR
Expose a new property in Public Client metadata and always registers the metadata endpoint (even when auth is not configured)

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/<number>

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ type ServerConfig struct {
 	DeprecatedThirdPartyConfig authConfig.ThirdPartyConfigOptions `json:"thirdPartyConfig" pflag:",Deprecated please use auth.appAuth.thirdPartyConfig instead."`
 
 	DataProxy DataProxyConfig `json:"dataProxy" pflag:",Defines data proxy configuration."`
+
+	ExportedDNS config.URL `json:"exportedDNS" pflag:",Defines the http endpoint the service is accessible at."`
```
Comment (EngHabu) [inline pkg/config/config.go:30]: What should I call this 😬, @wild-endeavor ?

Comment (kumare3) [reply pkg/config/config.go:30]: serviceEndpoint?

Comment (wild-endeavor) [reply pkg/config/config.go:30]: HTTP_URL? :)

Comment (wild-endeavor) [reply pkg/config/config.go:30]: Can we add a comment to explain under what circumstance this should be used?

File: pkg/server/service.go [hunk 2]
```diff
@@ -107,8 +107,8 @@ func newGRPCServer(ctx context.Context, pluginRegistry *plugins.Registry, cfg *c
 
 	configuration := runtime2.NewConfigurationProvider()
 	service.RegisterAdminServiceServer(grpcServer, adminservice.NewAdminServer(ctx, pluginRegistry, configuration, cfg.KubeConfig, cfg.Master, dataStorageClient, scope.NewSubScope("admin")))
+	service.RegisterAuthMetadataServiceServer(grpcServer, authCtx.AuthMetadataService())
```
Comment (katrogan) [inline pkg/server/service.go:110]: how does this work if UseAuth isn't true?

**Review by katrogan — DISMISSED** [review]

**Review by kumare3 — DISMISSED** [review]

**Pull request merged**